app/models/manageiq/providers/amazon/cloud_manager/metrics_capture.rb

```diff
@@ -1,66 +1,85 @@
+# frozen_string_literal: true
+
 class ManageIQ::Providers::Amazon::CloudManager::MetricsCapture < ManageIQ::Providers::BaseManager::MetricsCapture
-  INTERVALS = [5.minutes, 1.minute]
+  INTERVALS = [5.minutes, 1.minute].freeze
 
   COUNTER_INFO = [
     {
-      :amazon_counters       => ["CPUUtilization"],
+      :amazon_counters       => %w[CPUUtilization].freeze,
       :calculation           => ->(stat, _) { stat },
-      :vim_style_counter_key => "cpu_usage_rate_average"
-    },
+      :vim_style_counter_key => "cpu_usage_rate_average",
+    }.freeze,
 
     {
-      :amazon_counters       => ["DiskReadBytes", "DiskWriteBytes"],
+      :amazon_counters       => %w[DiskReadBytes DiskWriteBytes].freeze,
       :calculation           => ->(*stats, interval) { stats.compact.sum / 1024.0 / interval },
-      :vim_style_counter_key => "disk_usage_rate_average"
-    },
+      :vim_style_counter_key => "disk_usage_rate_average",
+    }.freeze,
 
     {
-      :amazon_counters       => ["NetworkIn", "NetworkOut"],
+      :amazon_counters       => %w[NetworkIn NetworkOut].freeze,
       :calculation           => ->(*stats, interval) { stats.compact.sum / 1024.0 / interval },
-      :vim_style_counter_key => "net_usage_rate_average"
-    },
-  ]
+      :vim_style_counter_key => "net_usage_rate_average",
+    }.freeze,
 
-  COUNTER_NAMES = COUNTER_INFO.collect { |i| i[:amazon_counters] }.flatten.uniq
+    # Memory counters setup: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/mon-scripts.html
```

Comment:
@AlexanderZagaynov we could probably fetch also the swap ? Like https://github.com/Ladas/manageiq-providers-openstack/blob/cbb3191e2ef0a28143b1d3339bd900cb6c3c7d72/app/models/manageiq/providers/openstack/infra_manager/metrics_capture.rb#L58

Seems to be available